src/test/java/GmailServletTest.java

```diff
@@ -101,22 +166,277 @@ public void setUp() throws IOException, GeneralSecurityException {
   }
 
   @Test
-  public void noMessagesPresent() throws IOException, ServletException {
-    Mockito.when(gmailClient.listUserMessages(DEFAULT_QUERY_STRING)).thenReturn(NO_MESSAGES);
+  public void noNDaysParameter() throws IOException, ServletException {
+    Mockito.when(request.getParameter(Mockito.eq("nDays"))).thenReturn(null);
+    Mockito.when(request.getParameter(Mockito.eq("mHours")))
+        .thenReturn(String.valueOf(DEFAULT_M_HOURS));
+
+    Mockito.when(gmailClient.listUserMessages(Mockito.anyString())).thenReturn(NO_MESSAGES);
+
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Mockito.verify(response, Mockito.times(1)).sendError(Mockito.eq(400), Mockito.anyString());
+  }
+
+  @Test
+  public void noMHoursParameter() throws IOException, ServletException {
+    Mockito.when(request.getParameter(Mockito.eq("nDays")))
+        .thenReturn(String.valueOf(DEFAULT_M_HOURS));
+    Mockito.when(request.getParameter(Mockito.eq("mHours"))).thenReturn(null);
+
+    Mockito.when(gmailClient.listUserMessages(Mockito.anyString())).thenReturn(NO_MESSAGES);
+
     servlet.doGet(request, response);
     printWriter.flush();
-    List<Message> messages = gson.fromJson(stringWriter.toString(), LIST_OF_MESSAGES_TYPE);
-    Assert.assertTrue(messages.isEmpty());
+    Mockito.verify(response, Mockito.times(1)).sendError(Mockito.eq(400), Mockito.anyString());
   }
 
   @Test
-  public void someMessagesPresent() throws IOException, ServletException {
-    Mockito.when(gmailClient.listUserMessages(DEFAULT_QUERY_STRING)).thenReturn(THREE_MESSAGES);
+  public void negativeNDaysParameter() throws IOException, ServletException {
+    Mockito.when(request.getParameter(Mockito.eq("nDays")))
+        .thenReturn(String.valueOf(NEGATIVE_N_DAYS));
+    Mockito.when(request.getParameter(Mockito.eq("mHours")))
+        .thenReturn(String.valueOf(DEFAULT_M_HOURS));
+
+    Mockito.when(gmailClient.listUserMessages(Mockito.anyString())).thenReturn(NO_MESSAGES);
+
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Mockito.verify(response, Mockito.times(1)).sendError(Mockito.eq(400), Mockito.anyString());
```

Comment:
Nit: It's preferred to test state than interactions. If `Mockito` makes that hard, a fake could be an option.

Reply:
I gave the fake a try - thanks for the idea! Let me know if it looks good/bad.